app/uk/gov/hmrc/registerncfstub/controllers/RegisterNcfController.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.registerncfstub.controllers
+
+import java.time.Instant
+import java.util.UUID
+
+import javax.inject.{Inject, Singleton}
+import play.api.libs.json._
+import play.api.mvc._
+import uk.gov.hmrc.play.bootstrap.controller.BackendController
+import uk.gov.hmrc.registerncfstub.config.AppConfig
+import uk.gov.hmrc.registerncfstub.model._
+import uk.gov.hmrc.registerncfstub.services.RegisterNcfService
+@Singleton
+class RegisterNcfController @Inject()(appConfig: AppConfig, registerNcfService: RegisterNcfService, cc: ControllerComponents)
+    extends BackendController(cc) {
+
+  def receiveNcfData: Action[JsValue] = Action(parse.json) { implicit request =>
+    request.body.validate[NcfRequestData] match {
+      case JsSuccess(t, _) =>
+        registerNcfService.processRegisterNcfRequest(t) match {
+          case CompletedSuccessfully(mrn, responseCode)   => Ok(Json.toJson(NcfResponse(mrn, responseCode, None)))
+          case TechnicalError(mrn, responseCode, e)       => BadRequest(Json.toJson(NcfResponse(mrn, responseCode, Some(e))))
```

Comment:
As discussed, the only query I had was do we need to mock a 500 response from EIS too?

(Mauva is due to provide an updated spec documenting HTTP headers in responses, which we can integrate into the stub later.)

Reply:
Updated to include that